### PR TITLE
Allow override of create's OnFailure keyword from ROLLBACK to DO_NOTHING

### PR DESCRIPTION
**Pull request by oazmon**

### CODE REVIEW

File: sceptre/stack.py [hunk 1]
```diff
@@ -196,7 +196,8 @@ def create(self):
             "Tags": [
                 {"Key": str(k), "Value": str(v)}
                 for k, v in self.config.get("stack_tags", {}).items()
-            ]
+            ],
+            "OnFailure": self.config.get("on_failure", 'ROLLBACK')
```
Comment (theseanything) [inline sceptre/stack.py]: I feel `OnFailure` should only be added to the kwargs when its present in the stack config. To ensure AWS CloudFormation's default behaviour occurs (which could change from `ROLLBACK`) - the `OnFailure` should be absent from the API call.

**Comment by oazmon** [comment]
Changes managed

File: sceptre/stack.py [hunk 2]
```diff
@@ -196,9 +196,10 @@ def create(self):
             "Tags": [
                 {"Key": str(k), "Value": str(v)}
                 for k, v in self.config.get("stack_tags", {}).items()
-            ],
-            "OnFailure": self.config.get("on_failure", 'ROLLBACK')
+            ]
         }
+        if self.config.get("on_failure"):
+            self.config['OnFailure'] = self.config.get("on_failure")
```
Comment (theseanything) [inline sceptre/stack.py]: Not sure why you've referenced `self.config['OnFailure'] =`?
Surely you want something like this:
`create_stack_kwargs.update({"OnFailure": self.config.get("on_failure")})`

File: sceptre/stack.py [hunk 3]
```diff
@@ -196,9 +196,10 @@ def create(self):
             "Tags": [
                 {"Key": str(k), "Value": str(v)}
                 for k, v in self.config.get("stack_tags", {}).items()
-            ],
-            "OnFailure": self.config.get("on_failure", 'ROLLBACK')
+            ]
         }
+        if self.config.get("on_failure"):
```
Comment (theseanything) [inline sceptre/stack.py]: the more pythonic way is -> `if "on_failure" in self.config:`

**Comment by oazmon** [comment]
2nd round fixes done

**Comment by theseanything** [comment]
Could we add unit tests for this?

**Comment by oazmon** [comment]
1. Round 3 of fixes.
2. Thanks for improving my Python (I seek Java and C natively)
3. I have been trying to reach someone at your company, and response. I work for Intuit Inc. and we are looking to use your tool company-wide. We want some additions. Could you please relay to the right person.

**Comment by theseanything** [comment]
Thanks @oazmon for you contribution! For any enquires generally about Sceptre you can reach out to sceptre@cloudreach.com. For bugs and feature requests you can raise an issue on GitHub [here](https://github.com/cloudreach/sceptre/issues).

**Pull request merged**